apiVersion: v1
kind: ServiceAccount
metadata:
  name: k8s-py-client
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: k8s-py-client
rules:
- apiGroups: [""]
  resources: ["pods", "pods/log", "pods/status"]
  verbs: ['*']
- apiGroups: [""]
  resources: ["namespaces", "events"]
  verbs: ["get", "watch", "list"]
---
kind: ClusterRoleBinding
apiVersion: rbac.authorization.k8s.io/v1
metadata:
  name: k8s-py-client
subjects:
- kind: ServiceAccount
  name: k8s-py-client
  namespace: default
roleRef:
  kind: ClusterRole
  name: k8s-py-client
  apiGroup: ""
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: k8s-py-client
  labels:
    app: k8s-py-client
spec:
  replicas: 1
  selector:
    matchLabels:
      app: k8s-py-client
  template:
    metadata:
      labels:
        app: k8s-py-client
    spec:
      serviceAccountName: k8s-py-client
      containers:
      - name: k8s-py-client
        image: andreistefanciprian/pod-restarter@sha256:442a9c2f745ed89651b11014db9cc13b220c7da14594112a4b22e91d0b229421
        command: ["/bin/sh"]
        args: ["-c", "python main.py --error-message='$(ERROR_MESSAGE)' --polling-interval=$(POLLING_INTERVAL)"]
        resources:
            limits:
              cpu: 100m
              memory: 128Mi
            requests:
              cpu: 100m
              memory: 128Mi
        env:
          - name: ERROR_MESSAGE
            value: Failed to pull image "wrongimage"
          - name: POLLING_INTERVAL
            value: "30"
          - name: KUBE_AUTH_INSIDE_CLUSTER
            value: "True"